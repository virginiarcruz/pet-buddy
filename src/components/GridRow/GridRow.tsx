import React, { useState, useEffect, useContext } from 'react'
import {
  Row,
  PaginationContainer,
  ListContainer,
  ContentGrid,
  TitleColumn,
  Column
} from './styles'
import Avatar from 'components/Avatar'
import Button from 'components/Button'
import AppContext from 'context/AppContext'

import { Pagination } from 'components/Pagination'

import MediaMatch from 'components/MediaMatch'
import Card from 'components/Card'
import PageHeader from 'components/PageHeader'

const GridRow = () => {
  const context = useContext(AppContext)
  const { profile } = context
  const [current, setCurrent] = useState(0)

  console.log('profile', context.profile)

  const [searchTerm, setSearchTerm] = useState('')
  const [searchResults, setSearchResults] = useState()

  useEffect(() => {
    const results = profile?.filter((item) =>
      item.Name?.toLowerCase().includes(searchTerm)
    )
    setSearchResults(results)
  }, [profile, searchTerm])

  function paginate(arr, size) {
    return arr?.reduce((acc, val, i) => {
      const idx = Math.floor(i / size)
      const page = acc[idx] || (acc[idx] = [])
      page.push(val)

      return acc
    }, [])
  }

  const pages = paginate(searchResults, 10)

  const onClickPage = (e) => {
    e.preventDefault()
    setCurrent(e.target.outerText - 1)
  }

  const handleChange = (e) => {
    setSearchTerm(e.target.value)
  }

  return (
    <>
      <PageHeader
        inputValue={searchTerm}
        onChange={handleChange}
        labelBadge={profile?.length}
      />
      <MediaMatch greaterThan="medium">
        <ContentGrid>
          <Column />
          <TitleColumn> Name </TitleColumn>
          <TitleColumn>Type</TitleColumn>
          <TitleColumn> Breed </TitleColumn>
          <TitleColumn> Gender </TitleColumn>
          <TitleColumn> Color </TitleColumn>
          <Column />
        </ContentGrid>
      </MediaMatch>
      <ListContainer className="petsList">
        <MediaMatch greaterThan="medium">
          {pages?.[`${current}`]?.map((pet, i) => (
            <Row key={`pet-${i}`}>
              <Avatar
                imageAlt={pet.name}
                imageSrc={`https://picsum.photos/50?random=${i + 1}`}
              />
              <p> {pet.Name ? pet.Name : 'no name'} </p>
              <p> {pet.Type?.Name} </p>
              <p> {pet.Breed?.Primary?.Name} </p>
              <p> {pet.Sex?.Name} </p>
              <p> {pet.Features?.PrimaryColour} </p>
              <Button as="a" href="#">
                Details
              </Button>
            </Row>
          ))}
        </MediaMatch>
        <MediaMatch lessThan="medium">
          {pages?.[`${current}`]?.map((pet, i) => (
            <Card
              key={`pet-${i}`}
              idCard={`card-${i}`}
              petName={pet.Name ? pet.Name : 'no name'}
              petType={pet.Type?.Name}
              petBreed={pet.Breed?.Primary?.Name}
              petGender={pet.Sex?.Name}
              petColor={pet.Features?.PrimaryColour}
            />
          ))}
        </MediaMatch>
      </ListContainer>
      <PaginationContainer>
        {pages?.map((_, i) => {
          return (
            <Pagination
              key={`page-${i}`}
              href="#"
              pageLabel={i + 1}
              onClick={onClickPage}
              isActive={i === current}
            />
          )
        })}
      </PaginationContainer>
    </>
  )
}
export default GridRow
