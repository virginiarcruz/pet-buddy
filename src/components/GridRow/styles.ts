import styled, { css } from 'styled-components'
import { Grid } from 'components/Grid'
import { customMedia } from 'components/MediaMatch'

export const ListContainer = styled.div`
  .card-container {
    &:not(:last-child) {
      margin-bottom: 16px;
    }
  }
`
export const TitleColumn = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.size.medium};
    font-weight: ${theme.font.medium};
    color: ${theme.color.darkGray};
    line-height: 19px;
    padding-bottom: 24px;
    text-align: left;
  `}
`

export const ContentGrid = styled(Grid)`
  padding: 0 16px;
  ${customMedia.greaterThan('tablet')`
    margin-top: 56px;
  `}
`

export const Column = styled.div``

export const Row = styled(Grid)`
  ${({ theme }) => css`
    background-color: ${theme.color.lighterGray};
    padding: 12px 16px;
    margin-bottom: 12px;
    height: 72px;
    text-align: left;
    align-items: center;

    p {
      font-size: ${theme.font.size.medium};
      color: ${theme.color.black};
      font-weight: ${theme.font.medium};
      line-height: 19px;
    }
  `}
`

export const PaginationContainer = styled.div`
  display: flex;
  margin-top: 32px;
  overflow-x: scroll;
  ${customMedia.greaterThan('tablet')`
    overflow: hidden;
  `}
`
