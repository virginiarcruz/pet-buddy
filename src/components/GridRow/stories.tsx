import { Story, Meta } from '@storybook/react/types-6-0'
import GridRow from '.'

export default {
  title: 'GridRow',
  component: GridRow
} as Meta

export const Default: Story = () => <GridRow />
