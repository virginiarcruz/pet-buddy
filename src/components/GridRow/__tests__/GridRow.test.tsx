import React from 'react'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import GridRow from '../GridRow'

describe('<GridRow />', () => {
  it('should render the <GridRow />', () => {
    const { container } = renderWithTheme(<GridRow />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
