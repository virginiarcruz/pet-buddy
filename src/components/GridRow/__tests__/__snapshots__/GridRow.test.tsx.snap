// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridRow /> should render the content 1`] = `
.c5 {
  display: none;
}

.c10 {
  display: none;
}

.c6 {
  display: grid;
  grid-template-columns: 1fr;
}

.c9 .card-container:not(:last-child) {
  margin-bottom: 16px;
}

.c8 {
  font-size: 16px;
  font-weight: 500;
  color: #707183;
  line-height: 19px;
  padding-bottom: 24px;
  text-align: left;
}

.c7 {
  padding: 0 16px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 32px;
  overflow-x: scroll;
}

.c3 {
  border-radius: 100px;
  background-color: #F47500;
  color: #FFFFFF;
  font-weight: 600;
  font-size: 14px;
  padding: 4px 8px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: 25px;
  line-height: 17px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: #000121;
  font-family: 'Inter',-apple-system,sans-serif;
  font-size: 22px;
  font-weight: 600;
  margin-right: 8px;
  line-height: 27px;
}

.c4 {
  padding: 12px 12px 12px 32px;
  background-color: #F4F4F7;
  background-image: url('img/icon-search.svg');
  background-repeat: no-repeat;
  background-position: 12px 14px;
  border-radius: 12px;
  border: 0;
  font-size: 16px;
  font-family: 'Inter',-apple-system,sans-serif;
  color: #A1A2AF;
  line-height: 19px;
  width: 100%;
}

.c4::-webkit-input-placeholder {
  color: #A1A2AF;
}

.c4::-moz-placeholder {
  color: #A1A2AF;
}

.c4:-ms-input-placeholder {
  color: #A1A2AF;
}

.c4::placeholder {
  color: #A1A2AF;
}

.c0 {
  margin-bottom: 32px;
}

.c0 > input {
  margin-top: 32px;
}

@media (min-width:992px) {
  .c5 {
    display: block;
  }
}

@media (max-width:992px) {
  .c10 {
    display: block;
  }
}

@media (min-width:992px) {
  .c6 {
    display: grid;
    grid-template-columns: 70px 1fr 1fr 1.8fr 0.8fr 1.5fr 65px;
    grid-gap: 10px;
  }
}

@media (min-width:992px) {
  .c7 {
    margin-top: 56px;
  }
}

@media (min-width:992px) {
  .c11 {
    overflow: hidden;
  }
}

@media (min-width:992px) {
  .c4 {
    width: 310px;
  }
}

@media (min-width:992px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    margin-bottom: 0;
  }

  .c0 > input {
    margin-top: 0;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Your Animals
      </h1>
      <span
        class="c3"
      />
    </div>
    <input
      class="c4"
      id="inputFilter"
      placeholder="Search an animal by name"
      type="text"
      value=""
    />
  </header>
  <div
    class="c5"
  >
    <div
      class="c6 c7"
      data-testid="header-desktop"
    >
      <div
        class=""
      />
      <h2
        class="c8"
      >
         Name 
      </h2>
      <h2
        class="c8"
      >
        Type
      </h2>
      <h2
        class="c8"
      >
         Breed 
      </h2>
      <h2
        class="c8"
      >
         Gender 
      </h2>
      <h2
        class="c8"
      >
         Color 
      </h2>
      <div
        class=""
      />
    </div>
  </div>
  <div
    class="c9 petsList"
  >
    <div
      class="c5"
    />
    <div
      class="c10"
    />
  </div>
  <div
    class="c11"
    data-testid="pagination"
  />
</div>
`
